cumulusci/cli/service.py

```diff
@@ -157,7 +157,7 @@ def callback(*args, **kwargs):
             validator_path = service_config.get("validator")
             if validator_path:
                 validator = import_global(validator_path)
-                validator(serv_conf)
+                serv_conf = validator(serv_conf)
```

Comment:
Good catch. This isn't actually safe since any validator returning `None` would clobber `serv_conf`. I'll add a conditional here.

Now, we don't actually have a `Service` class with a public API, so I'm uncertain whether it is worth calling out as a Critical Change... but no harm done by doing so.

I was tempted to add such a class here, but decided against it to limit the scope of the change. Given our experience with SFDX I'm coming around to a bolder approach.